main.cpp

```diff
@@ -115,19 +139,19 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T t1(9, "text1");                                             //6
+    T t2(3, "text2");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
+    T* f;                                            //7
+    auto* smaller = f->compare(5, 7);                              //8
```

Comment:
the `compare` function defined on line 72 has this syntax:
```
T* compare(T* a, T* b)
```
You need to pass it 2 pointers to instances of `T`.  You have 2 instances of `T` on lines 142 and 143.  Pass pointers to those instances to the compare function.
- [ ] revise